<?xml version="1.0"?>
<launch>
  <!-- <arg name="input/pointcloud" default="/concatenated/pointcloud"/> -->
  <arg name="input/pointcloud" default="/pc/lidar/top/pointcloud"/>
  <arg name="output/objects" default="/lidar_centerpoint_wit_objects"/>
  <arg name="model_name" default="default" description="options: `default` or `aip_x2`"/>
  <arg name="model_path" default="$(find-pkg-share lidar_centerpoint_wit)/data"/>
  <arg name="model_param_path" default="$(find-pkg-share lidar_centerpoint_wit)/config/$(var model_name).param.yaml"/>
 
  <node pkg="lidar_centerpoint_wit" exec="lidar_centerpoint_node" name="lidar_centerpoint_wit" output="screen">
    <remap from="~/input/pointcloud" to="$(var input/pointcloud)"/>
    <remap from="~/output/objects" to="$(var output/objects)"/>
    <param name="score_threshold" value="0.45"/>
    <param name="densification_world_frame_id" value="top_lidar"/>
    <param name="densification_num_past_frames" value="1"/>
    <param name="trt_precision" value="fp16"/>
    <param name="encoder_onnx_path" value="$(var model_path)/centerpoint_pillar_wit_pillar_encode.onnx"/>
    <param name="encoder_engine_path" value="$(var model_path)/centerpoint_pillar_wit_pillar.engine"/>
    <param name="head_onnx_path" value="$(var model_path)/centerpoint_pillar_wit_backbone_head.onnx"/>
    <param name="head_engine_path" value="$(var model_path)/centerpoint_pillar_wit_backbone_head.engine"/>
    <param from="$(var model_param_path)"/>
  </node>
</launch>
